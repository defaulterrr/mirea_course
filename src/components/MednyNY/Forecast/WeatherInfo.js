import React from 'react';
import PropTypes from 'prop-types';
import styles from './weather.module.css';

class WeatherInfo extends React.Component {
  constructor() {
    super();
    this.state = {
      weatherData: null,
    };
  }

  componentDidMount() {
    const { zip } = this.props;
    const apiKey = 'b1b35bba8b434a28a0be2a3e1071ae5b';
    const URL = `http://api.openweathermap.org/data/2.5/weather?q=${zip}&appid=${apiKey}&units=metric`;
    fetch(URL).then((res) => res.json()).then((json) => {
      this.setState({ weatherData: json });
    });
  }

  render() {
    const { weatherData } = this.state;
    if (!weatherData) { return <div>Loading</div>; }
    const weather = weatherData.weather[0];
    return (
      <div className={styles.mainDiv}>
        <form className={styles.form}>
          <div className={styles.img}>
            <img className={styles.weatherImage} alt="weather" src="https://i.pinimg.com/564x/1a/5e/15/1a5e153490c4fd46cb8d3a8254dbba48.jpg" />
          </div>
          <div className={styles.info}>
            <h1 className={styles.title}>
              {weather.main}
              {' '}
              in
              {' '}
              {weatherData.name}
            </h1>
            <p className={styles.parag}>
              Temprature:
              {(weatherData.main.temp).toFixed(0)}
              °С
            </p>
            <p className={styles.parag}>
              High Temp:
              {(weatherData.main.temp_max).toFixed(0)}
              °С
            </p>
            <p className={styles.parag}>
              Low Temp:
              {(weatherData.main.temp_min).toFixed(0)}
              °С
            </p>
            <p className={styles.parag}>
              Wind Speed:
              {(weatherData.wind.speed).toFixed(0)}
              {' '}
              mi/hr
            </p>
            <p className={styles.parag}>
              Humidity:
              {weatherData.main.humidity}
            </p>
          </div>
        </form>
      </div>
    );
  }
}

WeatherInfo.propTypes = {
  zip: PropTypes.string.isRequired,
};

export default WeatherInfo;
