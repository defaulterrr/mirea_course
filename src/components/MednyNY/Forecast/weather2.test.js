import React from 'react';
import { create } from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import WeatherInfo from './WeatherInfo';

configure({ adapter: new Adapter() });

const testData = {
  data: {
    base: 'stations',
    clouds: { all: 90 },

    cod: 200,
    coord:
    {
      lat: 55.75,
      lon: 37.62,
    },
    dt: 1591188032,
    id: 524901,
    main:
    {
      feels_like: 53.17,
      humidity: 100,
      pressure: 1009,
      temp: 57.2,
      temp_max: 59,
      temp_min: 55.4,
    },
    name: 'Moscow',
    sys:
    {
      country: 'RU',
      id: 9029,
      sunrise: 1591145467,
      sunset: 1591207478,
      type: 1,
    },

    timezone: 10800,
    visibility: 10000,
    weather:
    {
      0: {
        description: 'light rain',
        icon: '10d',
        id: 500,
        main: 'Rain',
      },
      length: 1,
    },

    wind:
{
  deg: 90,
  speed: 11.18,
},
  },
};

describe('Snapshot test', () => {
  it('Matches the snapshot', () => {
    const weatherInfo = create(<WeatherInfo weatherData={testData.data.runs} isLoading={false} />);
    expect(weatherInfo.toJSON()).toMatchSnapshot();
  });
});
