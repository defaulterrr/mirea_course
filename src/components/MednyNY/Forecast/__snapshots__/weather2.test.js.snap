// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Matches the snapshot 1`] = `
<div>
  Loading
</div>
`;
